mesonbuild/interpreter/type_checking.py

```diff
@@ -376,6 +377,13 @@ def _output_validator(outputs: T.List[str]) -> T.Optional[str]:
     default=[],
 )
 
+def include_dir_string_new(val: T.List[T.Union[str, IncludeDirs]], subproject: SubProject, node: mparser.BaseNode) -> None:
+    strs = [v for v in val if isinstance(v, str)]
+    if strs:
+        str_msg = ", ".join(f"'{s}'" for s in strs)
+        FeatureNew.single_use('include_directories kwarg of type string', '1.0.0', subproject,
+                              f'Use include_directories({str_msg}) instead', location=node)
```

Comment:
I wonder if maybe this should just 

```python
return FeatureNew('include_directories kwarg of type string', '1.0.0', f'Use include_directories({str_msg}) instead')
```

This would simplify the function signature for this helper quite a bit.

Reply:
But there's not reason you couldn't trigger multiple Features from the check here. I actually did this originally, but then realized that you'd have to accept a list of `FeatureNew` and `FeatureDeprecated` objects.

Reply:
Okay, I ended up doing this anyway, it's not a big deal to have the loop I guess.